packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -21,13 +22,17 @@ export class CheckboxBase extends React.Component<ICheckboxProps, ICheckboxState
   private _classNames: { [key in keyof ICheckboxStyles]: string };
 
   public static getDerivedStateFromProps(props: ICheckboxProps, state: ICheckboxState): ICheckboxState {
+    if (!props.defaultIndeterminate && state.isIndeterminate) {
+      return {
+        isIndeterminate: !!props.indeterminate
+      };
+    }
     if (props.checked !== undefined) {
       return {
-        ...state,
         isChecked: !!props.checked
       };
     }
-    return state;
+    return {};
```

Comment:
We should return null here--it might still trigger a state update (when it's not actually needed) if you use an empty object.

Reply:
Just update the return type of the method to include `| null`